src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/RerankingMetrics.java

```diff
@@ -0,0 +1,56 @@
+package io.stargate.sgv2.jsonapi.service.operation.reranking;
+
+import static io.stargate.sgv2.jsonapi.util.ClassUtils.classSimpleName;
+
+import io.micrometer.core.instrument.*;
+import io.stargate.sgv2.jsonapi.api.request.RequestContext;
+import io.stargate.sgv2.jsonapi.service.reranking.operation.RerankingProvider;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class RerankingMetrics {
```

Comment:
doc comments pls:  on the class and on the public functions that are tracking metrics